usecase/parser/utils/parsermanager.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/crypto-com/chain-indexing/entity/command"
+	applogger "github.com/crypto-com/chain-indexing/internal/logger"
+	"github.com/crypto-com/chain-indexing/usecase/event"
+	"github.com/crypto-com/chain-indexing/usecase/model"
+)
+
+type CosmosParserManager struct {
+	store  map[string]CosmosParserBlockHeight
+	logger applogger.Logger
+	config CosmosParserManagerConfig
+}
+
+type CosmosParserBlockHeight map[uint64]CosmosParser
+
+type CosmosParserManagerConfig struct {
+	CosmosVersionBlockHeight
+}
+
+type CosmosVersionBlockHeight struct {
+	V0_43_0 uint64
+}
+
+type CosmosParserManagerParams struct {
+	Logger applogger.Logger
+	Config CosmosParserManagerConfig
+}
+
+type CosmosParser func(
+	CosmosParserParams,
+) []command.Command
+
+type CosmosParserParams struct {
+	MsgCommonParams event.MsgCommonParams
+	TxsResult       model.BlockResultsTxsResult
+	MsgIndex        int
+	Msg             map[string]interface{}
+}
+
+func NewCosmosParserManager(params CosmosParserManagerParams) *CosmosParserManager {
+	cpm := &CosmosParserManager{
+		store:  make(map[string]CosmosParserBlockHeight),
+		logger: params.Logger,
+		config: params.Config,
+	}
+
+	return cpm
+}
+
+func (cpm *CosmosParserManager) GetCosmosV0_43_0BlockHeight() uint64 {
+	return cpm.config.CosmosVersionBlockHeight.V0_43_0
+}
+
+func (cpm *CosmosParserManager) RegisterParser(name string, fromHeight uint64, parser CosmosParser) {
+	if cpm.store[name] == nil {
+		cpm.store[name] = make(map[uint64]CosmosParser)
+	}
```

Comment:
To avoid future mistake, can add a panic if the message and starting height already have a parser registered.

Reply:
But it should be ok to override the registered parser i.e. 0

Reply:
Oh I see, that make senses.